### DISCUSSION TITLE
Emitted Event in Anvil test environment

### DISCUSSION THREAD

**Discussion by Aashim-Limbu** [Cyfrin/foundry-full-course-cu#3126] [Q&A]
### **Description**
Hey folks, I'm working on a personal project and encountered an issue while listening to events on the Anvil chain.  

The event is not being emitted when I run the contract on the local Anvil chain. However, when I switch to a test network, I can see the event being triggered and reflected in my frontend as well.  

### **Question**
Do events do not get emitted on the local Anvil chain?  

![image](https://github.com/user-attachments/assets/6cfbf57e-b6ab-431a-adad-e0cc6bfb42b6)

**Comment by EngrPips** [comment]
Hello @Aashim-Limbu, I am unsure how events work on Anvil, but honestly, I will consider that events are not properly stored for querying or listening since Anvil is just a one-time blockchain for quick use so it makes sense that it doesn't properly persist some data or maybe expose it as necessary.

**Reply by Aashim-Limbu** [reply]
@EngrPips  Oh indeed that make sense 🤔

**Reply by EngrPips** [reply]
Yeah

**Reply by Aashim-Limbu** [reply]
@EngrPips I have found out taht the event emitted from the Anvil chain can be listen into . you just want to create a contract instance as
---
![image](https://github.com/user-attachments/assets/c3c934a6-f01f-47aa-bd73-734adc40231d)
---
![image](https://github.com/user-attachments/assets/90195b62-7a26-4533-bc14-205a817d6bdc)
---
***Solution***
- You just need to listen to websocket connection , previously i tried to listen to http connection which was wrong as it only maintain connection till response is recieved .

**Reply by EngrPips** [reply]
Thanks for this heads-up @Aashim-Limbu, I truly appreciate it, but what do you mean by `it only maintain connection till response is recieved` when you were discussing about HTTP
